readthedocs/search/serializers.py

```diff
@@ -0,0 +1,143 @@
+"""
+Serializers for the ES's search result object.
+
+.. note::
+   Some fields are re-named to make their meaning more clear.
+   They should be renamed in the ES index too.
+"""
+
+import itertools
+from operator import attrgetter
+
+from django.shortcuts import get_object_or_404
+from rest_framework import serializers
+
+from readthedocs.core.resolver import resolve
+from readthedocs.projects.models import Project
+
+
+class ProjectHighlightSerializer(serializers.Serializer):
+
+    name = serializers.ListField(child=serializers.CharField(), default=list)
+    slug = serializers.ListField(child=serializers.CharField(), default=list)
+    description = serializers.ListField(child=serializers.CharField(), default=list)
+
+
+class ProjectSearchSerializer(serializers.Serializer):
+
+    type = serializers.CharField(default='project', source=None, read_only=True)
+    name = serializers.CharField()
+    slug = serializers.CharField()
+    link = serializers.CharField(source='url')
+    highlight = ProjectHighlightSerializer(source='meta.highlight', default=dict)
+
+
+class PageHighlightSerializer(serializers.Serializer):
+
+    title = serializers.ListField(child=serializers.CharField(), default=list)
+
+
+class PageSearchSerializer(serializers.Serializer):
+
+    type = serializers.CharField(default='page', source=None, read_only=True)
+    project = serializers.CharField()
+    version = serializers.CharField()
+    title = serializers.CharField()
+    path = serializers.CharField(source='full_path')
```

Comment:
I'm using just path to refer to the full path, we don't use the original "path" value (this is a sphinx page) in search results anymore. We maybe can drop this btw, we have the link attribute which contains the path too.

Reply:
This makes sense 👍